main.rb

```diff
@@ -0,0 +1,186 @@
+    if instance_of?(Array) || instance_of?(Range)
+      new_self.length.times do
+        yield(new_self[i])
+        i += 1
+      end
+    elsif instance_of?(Hash)
+      new_self.length / 2.times do
+        yield(new_self[i], new_self[i + 1])
+        i += 2
+      end
+    end
+    new_self
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    new_self = self
+    i = 0
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i], i)
+      i += 1
+    end
+    new_self
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    i = 0
+    arr = []
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    new_self.length.times do
+      yield(new_self[i]) == true ? arr = arr.push(new_self[i]) : arr
+      i += 1
+    end
+    arr
+  end
+
+  def my_all?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check == new_self.length
+  end
+
+  def my_any?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? true : false
+  end
+
+  def my_none?(parameter = nil)
+    i = 0
+    check = 0
+    new_self = self
+    new_self = new_self.instance_of?(Array) || new_self.instance_of?(Hash) ? flatten : to_a
+    while block_given? && i < new_self.length
+      yield(new_self[i]) == true ? check += 1 : check = check
+      i += 1
+    end
+    if parameter.instance_of?(Regexp)
+      new_self.length.times do
+        new_self[i] =~ parameter ? check += 1 : check = check
+        i += 1
+      end
+    else
+      new_self.length.times do
+        new_self[i].class == parameter ? check += 1 : check = check
+        i += 1
+      end
+    end
+    check.positive? ? false : true
+  end
```

Comment:
# my_none?     
- [ ] `my_none?` when no block or argument is given returns false only if one of the collection members is true

      Failure/Error: expect(true_array.my_none?).to be false

        expected false
             got true
##
- [ ] `my_none?` when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern

      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)

        expected false
             got true